web/src/components/snapshots/SnapshotStorageDestination.tsx

```diff
@@ -77,36 +77,88 @@ type State = {
   tmpFileSystemNFSServer?: string;
   tmpFileSystemType?: string;
   updatingSettings?: boolean;
-  useIamAws?: boolean;
+  useIamAws: boolean;
 };
 
 type StoreProviderName = "aws" | "gcp" | "azure" | "other";
-type StoreProvider = {
-  [K in StoreProviderName]?: {
-    accessKeyID?: string;
+
+type AWSStoreProvider = {
+  aws: {
+    region: string;
+    accessKeyID: string;
+    secretAccessKey: string;
+    useInstanceRole: boolean;
+  };
+  gcp?: undefined;
+  azure?: undefined;
+  other?: undefined;
+  path?: string;
+  bucket?: string;
+};
+
+type GCPStoreProvider = {
+  gcp: {
+    jsonFile: string;
+    serviceAccount: string;
+    useInstanceRole: boolean;
+  };
+  bucket?: undefined;
+  aws?: undefined;
+  azure?: undefined;
+  other?: undefined;
+  path?: undefined;
+};
+
+type AzureStoreProvider = {
+  azure: {
+    clientId: string;
+    clientSecret: string;
+    cloudName: string;
+    resourceGroup: string;
+    storageAccount: string;
+    subscriptionId: string;
+    tenantId: string;
+  };
+  bucket?: undefined;
+  aws?: undefined;
+  gcp?: undefined;
+  other?: undefined;
+  path?: undefined;
+};
+
+type OtherStoreProvider = {
+  other: {
+    region: string;
+    accessKeyID: string;
     accessKeySecret?: string;
-    clientId?: string;
-    clientSecret?: string;
-    cloudName?: string;
-    endpoint?: string;
-    jsonFile?: string;
-    region?: string;
-    resourceGroup?: string;
-    secretAccessKey?: string;
-    serviceAccount?: string;
-    storageAccount?: string;
-    subscriptionId?: string;
-    tenantId?: string;
-    useInstanceRole?: boolean;
+    secretAccessKey: string;
+    endpoint: string;
   };
+  bucket?: undefined;
+  aws?: undefined;
+  gcp?: undefined;
+  azure?: undefined;
+  path?: undefined;
 };
+
 type StoreMetadata = {
+  aws?: undefined;
+  gcp?: undefined;
+  azure?: undefined;
+  other?: undefined;
   bucket?: string;
   internal?: boolean;
   fileSystem?: string;
   path?: string;
 };
```

Comment:
These types are pretty messy. We could combine these into a single nomenclature and use a discriminated union type to simplify this. Future task. 

https://css-tricks.com/typescript-discriminated-unions/